apps/build-onchain-apps/pages/api/chains/supported.ts

```diff
@@ -0,0 +1,18 @@
+import { getChainsForEnvironment } from '../../../src/utils/chainConfiguration';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+/**
+ * Handler for the /api/chains/supported route, this route will return all the supported
+ * chains for this application.
+ * @param req
+ * @param res
+ */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const chains = getChainsForEnvironment();
+    res.status(200).json(chains);
+  } catch (error) {
+    console.error('Error fetching chains:', error);
+    res.status(500).json({ error: 'Internal server error' });
```

Comment:
Should we pass the error as well, in the response?

Reply:
Typically you don't want to pass along errors in production use-cases (as it could expose an attack surface, e.g. database error).  I do think it is reasonable to pass the error along in dev/localhost